src/components/PopupWithForm.js

```diff
@@ -29,4 +32,12 @@ export default class PopupWithFrom extends Popup {
     this._popupForm.reset();
     super.close();
   }
+
+  showLoading() {
+    this._submitButton.textContent = this._loadingButtonText;
+  }
+
+  hideLoading() {
+    this._submitButton.textContent = this._buttonText;
+  }
 }
```

Comment:
Please, delete `this.close();` from line 27  in `PopupWithForm.js`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well